src/core/mac/mac_types.hpp

```diff
@@ -499,72 +500,72 @@ class ExtendedPanId : public otExtendedPanId
 } OT_TOOL_PACKED_END;
 
 /**
- * This structure represents an IEEE802.15.4 Network Name.
+ * This class represents a char array as Data (pointer to a char buffer along with a length).
+ *
+ * @note The char array does NOT need to be null terminated.
  *
  */
-class NetworkName : public otNetworkName
+class Data
```

Comment:
note: here moves original `class NetworkName::Data` out as `class Data` which could be used commonly by `NetworkName` and `DomainName`.

Reply:
How about something more descriptive than `Data`? Maybe `Name?`

Reply:
Considered another `NameData` at the beginning. 
`NameData` or `Name`, which one do you prefer?

Reply:
`NameData` works for me.

Reply:
updated.